### ISSUE TITLE
Instance of 'ManyToManyField' has no 'all' memberPylint E1101:(no-member)

### ISSUE THREAD

**Issue by LCH-1** [pylint-dev/pylint-django#404]
```python
class Category(models.Model):
    name = models.CharField("name", max_length=100)

class Movie(models.Model):
    categories = models.ManyToManyField(Category, related_name="movies")

    def get_categories(self):
        return [category.name for category in self.categories.all()]
```

This code will cause an `Instance of 'ManyToManyField' has no 'all' memberPylint E1101:(no-member)` pylint error.

and same error occurs when using using self.categories.count(), etc

```
Python 3.11.1
Django==4.2.4
pylint==2.17.5
pylint-django==2.5.3
astroid==2.15.6
```

**Comment by LCH-1** [comment]
this error does not occur in terminal, but only in vscode pylint extension.

**Comment by Pierre-Sassoulas** [comment]
What are the versions used in the vscode extension ? You also need to check if you're in the same virtual environnement.

**Comment by LCH-1** [comment]
More detailed results were reported in the first comment.

I also checked that the selected virtual environment settings are the same as terminal and vscode selected interpreter.

**Comment by sjdemartini** [comment]
I'm experiencing the same problem, but I'm guessing it's an issue with the VSCode Extension (https://github.com/microsoft/vscode-pylint) and perhaps not specific to pylint-django. Seemingly the VSCode extension is (as of recently) failing to load the pylint-django plugin properly, regardless of whether I specify it with the `pylint.args` setting or in my `.pylintrc`. I'm not sure where to dig in further.

For what it's worth, I do see this near the top of the VSCode Pylint logs:
```
2024-04-17 14:44:36.710 [info] [Error - 2:44:36 PM] Traceback (most recent call last):
  File "<frozen runpy>", line 148, in _get_module_details
  File "<frozen runpy>", line 142, in _get_module_details
ImportError: No module named pylint.__main__

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sdemartini/.vscode/extensions/ms-python.pylint-2023.10.1/bundled/tool/lsp_server.py", line 770, in _run_tool_on_document
    result = utils.run_module(
             ^^^^^^^^^^^^^^^^^
  File "/Users/sdemartini/.vscode/extensions/ms-python.pylint-2023.10.1/bundled/tool/lsp_utils.py", line 208, in run_module
    return _run_module(module, argv, use_stdin, source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sdemartini/.vscode/extensions/ms-python.pylint-2023.10.1/bundled/tool/lsp_utils.py", line 193, in _run_module
    runpy.run_module(module, run_name="__main__")
  File "<frozen runpy>", line 222, in run_module
  File "<frozen runpy>", line 152, in _get_module_details
ImportError: No module named pylint.__main__; 'pylint' is a package and cannot be directly executed
```

**Comment by poulposse** [comment]
I guess we did not get the same issue. But I agree with you that it does not seem to be a pylint-django issue, but rather a pylint or VSCode extension issue.
In my case, the behavior was the following: the output of the very first call to pylint was correct, then the subsequent outputs were giving false positives (as if pylint was run without django plugin), although the call was exactly the same.

It turns out that the `--clear-cache-post-run` argument of pylint (that is added to my arguments by VSCode extension) is responsible of this behavior. I commented it out in `lsp_server.py` (function `_linting_helper`) in the VSCode extension, and 
it now behaves well.

(VSCode extension: v2023.10.1, pylint: 3.0.2, pylint-django: 4.5.2)
